app/main.py

```diff
@@ -1,6 +1,55 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    """
+    decorating only functions that take immutable arguments.
+    cache should print "Getting from cache" when returns stored value
+    and "Calculating new result" when run function with new arguments.
+
+    structure for DB_cache:
+        db_cache =
+        {   "function's name1":
+                {   (tuple1 of arguments): result1,
+                    (tuple2 of arguments): result2,
+                    ...
+                    (tupleN of arguments): resultN,
+                }
+            "function's name2": {...}
+            "function's nameN": {...}
+        }
+    """
+    # 0. initialise DB_cache
+    db_cache = {}
+
+    def wrapper(*func_args, **kwargs) -> Any:
+        # 1. check for immutable arguments, if not return func without caching
```

Comment:
delete all comments, if you need explanation just write docstring

Reply:
ok, deleted